.github/buildomat/jobs/build-and-test.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+#:
+#: name = "illumos / build-and-test"
+#: variety = "basic"
+#: target = "helios"
+#: rust_toolchain = "nightly"
+#: output_rules = []
+#:
+
+set -o errexit
+set -o pipefail
+set -o xtrace
+
+cargo --version
+rustc --version
+
+export RUST_BACKTRACE=1
+
+banner test
+ptime -m cargo test --release --no-fail-fast --verbose --workspace \
+    --features no-linker
```

Comment:
Apologies for the confusion about this feature. It's not really documented, and intentionally so. It was intended to provide developers a knob to compile and test cross-platform bits, rather than as a true public feature flag. It's definitely not required to build the fully-functional crate on illumos systems. They're kinda difficult to follow, but [this set of conditional-compilation directives](https://github.com/oxidecomputer/usdt/blob/014f85ac97d0d923c17e5dcd51b9293e51acef7a/usdt-impl/src/lib.rs#L16-L25) results in using the `no-linker.rs` implementation file on illumos, even without specifying the actual flag.